### PR TITLE
Adding support for Google Authenticator

### PR DESCRIPTION
**Pull request by whoot**
Gives the possibility for a user to choose between mOTP and Google Authenticator in Freeradius, when using one-time-passwords.

### CODE REVIEW

**Comment by whoot** [comment]
Something missing for merge?

File: net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc [hunk 1]
```diff
@@ -95,7 +95,14 @@ function freeradius_install_command() {
 		log_error("FreeRADIUS: Creating backup of the original file to " . FREERADIUS_ETC . "/raddb/files.backup");
 		copy(FREERADIUS_ETC . "/raddb/modules/files", FREERADIUS_ETC . "/raddb/files.backup");
 	}
-
+	
+	// Install Google Authenticator scripts
+	if (!file_exists(FREERADIUS_ETC . "/raddb/scripts/googleauth.py")) { 
+		copy(FREERADIUS_BASE . "/pkg/googleauth.py", FREERADIUS_ETC . "/raddb/scripts/"); 
+		exec("chmod +x " . FREERADIUS_ETC . "/raddb/scripts/googleauth.py");
```
Comment (rbgarga) [inline net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc]: PHP provides a function to chmod(), consider using it

File: net/pfSense-pkg-freeradius2/files/usr/local/pkg/googleauth.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#!/usr/local/bin/python2.7
```
Comment (rbgarga) [inline net/pfSense-pkg-freeradius2/files/usr/local/pkg/googleauth.py]: Is this code yours? It's missing license and copyright

Comment (whoot) [inline net/pfSense-pkg-freeradius2/files/usr/local/pkg/googleauth.py]: Some code is from here:
http://www.brool.com/post/using-google-authenticator-for-your-website/

License is  "CC0 1.0 Universal License"

File: net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc [hunk 3]
```diff
@@ -95,7 +95,14 @@ function freeradius_install_command() {
...
 		copy(FREERADIUS_ETC . "/raddb/modules/files", FREERADIUS_ETC . "/raddb/files.backup");
 	}
-
+	
+	// Install Google Authenticator scripts
+	if (!file_exists(FREERADIUS_ETC . "/raddb/scripts/googleauth.py")) { 
+		copy(FREERADIUS_BASE . "/pkg/googleauth.py", FREERADIUS_ETC . "/raddb/scripts/"); 
+		chmod(REERADIUS_ETC . "/raddb/scripts/googleauth.py",0755);
+	}
+	if (!file_exists(FREERADIUS_ETC . "/raddb/modules/googleauth")) { copy(FREERADIUS_BASE . "/pkg/googleauth", FREERADIUS_ETC . "/raddb/modules/");}
```
Comment (rbgarga) [inline net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc]: Please break this long line as is done on block just above it

**Comment by whoot** [comment]
Hey there, 
could you please help me out with the changes you requested?

Greetings

**Comment by whoot** [comment]
The latest development branch breaks the GoogleAuth support, because the requirement for the Init-Secret is "**Note: May only contain 0-9 and a-f. Must contain at least 16 characters.**" !
Please consider removing this restriction!

**Pull request closed without merging**